File: problems/sqrt/sqrt.py
class Solution:
    def mySqrt(self, x: int) -> int:
        if x <= 1:
            return x
        lo = 1
        hi = x
        while lo <= hi:
            mid = (lo + hi) // 2
            if mid * mid == x:
                return mid
            elif mid * mid < x:
                lo = mid + 1
            else:
                hi = mid - 1
        print(f"lo = {lo} and hi = {hi}")
        return hi

File: problems/sqrt/Sqrt.java
package problems.sqrt;

public class Sqrt {

    public int sqrt(int x) {
        if (x <= 1)
            return x;

        int lo = 1;
        int hi = x;

        while (lo <= hi) {
            int mid = (lo + hi) / 2;
            int sq = mid * mid;
            if (sq == x) {
                return mid;
            } else if (sq > x) {
                hi = mid - 1;
            } else {
                lo = mid + 1;
            }
        }

        return hi;
    }

}
